tests/unit/src/test/scala/tests/CompilersLspSuite.scala

```diff
@@ -0,0 +1,52 @@
+package tests
+
+import scala.concurrent.Future
+
+class CompilersLspSuite extends BaseCompletionLspSuite("compilers") {
+  test("reset-pc") {
+    cleanWorkspace()
+    for {
+      _ <- server.initialize(
+        """/metals.json
+          |{
+          |  "a": {},
+          |  "b": { "dependsOn": ["a"] }
+          |}
+          |/a/src/main/scala/a/A.scala
+          |package a
+          |object A {
+          |  // @@
+          |  def completeThisUniqueName() = 42
+          |}
+          |/b/src/main/scala/b/B.scala
+          |package b
+          |object B {
+          |  // @@
+          |  def completeThisUniqueName() = 42
+          |}
+          |""".stripMargin
+      )
+      _ <- server.didOpen("a/src/main/scala/a/A.scala")
+      _ <- server.didOpen("b/src/main/scala/b/B.scala")
+      _ = assertNoDiagnostics()
+      _ <- Future.sequence(
+        List('a', 'b').map(project =>
+          assertCompletion(
+            "completeThisUniqueNa@@",
+            "completeThisUniqueName(): Int",
+            project = project
+          )
+        )
+      )
+      _ = assertEquals(2, server.server.loadedPresentationCompilerCount())
+      _ <- server.didSave("b/src/main/scala/b/B.scala")(_ =>
+        "package b; object B"
+      )
+      _ <- server.didSave("a/src/main/scala/a/A.scala")(_ =>
+        "package a; object A"
+      )
+      _ = assertNoDiagnostics()
+      _ = assertEquals(0, server.server.loadedPresentationCompilerCount())
```

Comment:
Shouldn't we try to change both files to contain new names and assert that both new names are picked up from another file after only compiling `a`?

I think that would test the same thing and not require the additional `loadedPresentationCompilerCount`

Reply:
I agree with the principle that we should reproduce the user-facing situation instead of testing against implementation details. My concern with reproducing the exact scenario is that we'll end up with a flaky test. 

It's critical that we reset the presentation compiler for getting good user experience and I think it will be helpful in the future to have `isLoaded()` as a toolkit to test related fixes.